weave

```diff
@@ -1393,51 +1425,17 @@ launch_router() {
         --http-addr $HTTP_ADDR \
         --status-addr $STATUS_ADDR \
         --resolv-conf "/var/run/weave/etc/$RESOLV_CONF_BASE" \
-        "$@")
+        $PROXY_ARGS)
     wait_for_status $CONTAINER_NAME http_call $HTTP_ADDR
+    # Wait for Weave Proxy:
+    [ -n "$NO_PROXY" ] || wait_for_status $CONTAINER_NAME http_call_unix $CONTAINER_NAME status.sock
```

Comment:
I will remove this line entirely then. :+1: 

For the record, we have such a `wait_for_status` call in [`master`](https://github.com/weaveworks/weave/blob/master/weave#L1435) at the moment, hence I kept it and moved it here as I merged the router and the proxy. I added a condition given it makes no sense to wait if the proxy is disabled.